src/main/java/org/broadinstitute/consent/http/db/DatasetDAO.java

```diff
@@ -239,6 +239,185 @@ LEFT JOIN file_storage_object fso ON (fso.entity_id = d.dataset_id::text OR fso.
+              u.user_id AS u_user_id, u.email AS u_email, u.display_name AS u_display_name,
+              u.create_date AS u_create_date, u.email_preference AS u_email_preference,
+              u.institution_id AS u_institution_id, u.era_commons_id AS u_era_commons_id,
+              k.key, dp.property_value, dp.property_key, dp.property_type, dp.schema_property, dp.property_id,
+              ca.consent_id, c.translated_use_restriction,
+              s.study_id AS s_study_id,
+              s.name AS s_name,
+              s.description AS s_description,
+              s.data_types AS s_data_types,
+              s.pi_name AS s_pi_name,
+              s.create_user_id AS s_create_user_id,
+              s.create_date AS s_create_date,
+              s.update_user_id AS s_user_id,
+              s.update_date AS s_update_date,
+              s.public_visibility AS s_public_visibility,
+              s_dataset.dataset_id AS s_dataset_id,
+              sp.study_property_id AS sp_study_property_id,
+              sp.study_id AS sp_study_id,
+              sp.key AS sp_key,
+              sp.value AS sp_value,
+              sp.type AS sp_type,
+              fso.file_storage_object_id AS fso_file_storage_object_id,
+              fso.entity_id AS fso_entity_id,
+              fso.file_name AS fso_file_name,
+              fso.category AS fso_category,
+              fso.gcs_file_uri AS fso_gcs_file_uri,
+              fso.media_type AS fso_media_type,
+              fso.create_date AS fso_create_date,
+              fso.create_user_id AS fso_create_user_id,
+              fso.update_date AS fso_update_date,
+              fso.update_user_id AS fso_update_user_id,
+              fso.deleted AS fso_deleted,
+              fso.delete_user_id AS fso_delete_user_id
+          FROM dataset d
+          LEFT JOIN users u on d.create_user_id = u.user_id
+          LEFT JOIN (SELECT DISTINCT dataset_id AS id FROM dar_dataset) dar_ds_ids ON dar_ds_ids.id = d.dataset_id
+          LEFT JOIN dataset_property dp ON dp.dataset_id = d.dataset_id
+          LEFT JOIN dictionary k ON k.key_id = dp.property_key
+          LEFT JOIN consent_associations ca ON ca.dataset_id = d.dataset_id
+          LEFT JOIN study s ON s.study_id = d.study_id
+          LEFT JOIN study_property sp ON sp.study_id = s.study_id
+          LEFT JOIN dataset s_dataset ON s_dataset.study_id = s.study_id
+          LEFT JOIN file_storage_object fso ON (fso.entity_id = d.dataset_id::text OR fso.entity_id = s.uuid::text) AND fso.deleted = false
+          LEFT JOIN consents c ON c.consent_id = ca.consent_id
+          WHERE
+            (
+              (s.public_visibility IS NULL OR s.public_visibility = TRUE)
+              AND d.active = TRUE
+              AND d.dac_approval = TRUE
+            )
+            OR
+            (
+              s.create_user_id = :userId
+              OR (sp.key = 'dataCustodianEmail' AND sp.value LIKE concat('%"', :email, '"%'))
+            )
```

Comment:
Since the main difference between these queries is the where clause, I wonder if we should abstract that part away so it can be reused by these queries.